Classes/Private/Manager/Submanagers/OCTSubmanagerCalls.m

```diff
@@ -246,11 +236,45 @@ - (void)updateCall:(OCTCall *)call withStatus:(OCTCallStatus)status
     OCTRealmManager *realmManager = [self.dataSource managerGetRealmManager];
 
     [realmManager updateObject:call withBlock:^(OCTCall *callToUpdate) {
-
         callToUpdate.status = status;
+    }];
+}
+
+- (void)pause:(BOOL)pause controlSentToCall:(OCTCall *)call
+{
+    OCTRealmManager *realmManager = [self.dataSource managerGetRealmManager];
+
+    BOOL wasPaused = call.pausedStatus != OCTCallPausedStatusNone;
+    OCTCallPausedStatus previousPausedStatus = call.pausedStatus;
 
-        callToUpdate.onHoldStartInterval = (status == OCTCallStatusPaused) ? [[NSDate date] timeIntervalSince1970] : 0;
+    [realmManager updateObject:call withBlock:^(OCTCall *callToUpdate) {
+
+        if (pause && ! wasPaused) {
+            callToUpdate.onHoldStartInterval = [[NSDate date] timeIntervalSince1970];
+        }
+
+        if (pause) {
+            callToUpdate.pausedStatus |= OCTCallPausedStatusByUser;
+        }
+        else {
+            callToUpdate.pausedStatus &= ~OCTCallPausedStatusByUser;
+        }
     }];
+
+    OCTFriend *friend = [call.chat.friends firstObject];
+
+    if (pause && ! wasPaused) {
+        [self.timer stopTimer];
+        [self.audioEngine stopAudioFlow:nil];
+    }
+    else if (! pause &&  (previousPausedStatus == OCTCallPausedStatusByUser)) {
+        [self.audioEngine startAudioFlow:nil];
+        [self.timer startTimerForCall:call];
+        self.audioEngine.friendNumber = friend.friendNumber;
+    }
+    else if (! pause) {
+        self.audioEngine.friendNumber = friend.friendNumber;
+    }
```

Comment:
Tried to refactor this method a bit to reduce number of `else if` and used variables in the code.

```
- (void)putOnPause:(BOOL)pause call:(OCTCall *)call
{
    OCTRealmManager *realmManager = [self.dataSource managerGetRealmManager];
    BOOL wasPaused = call.pausedStatus != OCTCallPausedStatusNone;

    if (pause) {
        if (! wasPaused) {
            [self.timer stopTimer];
            [self.audioEngine stopAudioFlow:nil];
        }
    }
    else {
        OCTFriend *friend = [call.chat.friends firstObject];
        self.audioEngine.friendNumber = friend.friendNumber;

        if (call.pausedStatus & OCTCallPausedStatusByUser) {
            [self.audioEngine startAudioFlow:nil];
            [self.timer startTimerForCall:call];
        }
    }

    [realmManager updateObject:call withBlock:^(OCTCall *callToUpdate) {
        if (pause) {
            callToUpdate.pausedStatus |= OCTCallPausedStatusByUser;
            callToUpdate.onHoldStartInterval = callToUpdate.onHoldStartInterval ?: [[NSDate date] timeIntervalSince1970];
        }
        else {
            callToUpdate.pausedStatus &= ~OCTCallPausedStatusByUser;
            callToUpdate.onHoldStartInterval = nil;
        }
    }];
}
```